### PR TITLE
Data Science with Python : Regularization in Regression #4376

### LINKED ISSUE #4376: Data Science with Python : Regularization in Regression
### Description

Welcome to 'DSWP' Team, good to see you here

With this issue, readers will get introduced to core information to Regularization in Regression along with sample code completely in an effective way.

To get assigned to this issue, add your serial numbers mentioned in the spreadsheet of "Data Science with Python", the approach one would follow and the choice you prefer (Documentation, Audio, Video). You can go with all three or any number of options you're interested to work on.

If you had referred any resources, add them up in "DS Resources". Similarly, if you had used datasets, include them in "DS Datasets".

Domain: Machine Learning 

Mentors Assigned: Dhruv Bajaj / Aishani Singh 

Points to Note :

The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.

"Issue Title" and "PR Title should be the same. Include the issue number along with it.

Changes should be made inside the Datascience_With_Python/ directory & Datascience_With_Python branch.

Follow Contributing Guidelines & Code of Conduct before start Contributing.

This issue is only for 'GWOC' contributors of the 'DSWP' domain.

All the best. Enjoy your open source journey ahead. 😎

### Domain

Datascience with Python

### Type of Contribution

Video, Documentation

### Code of Conduct

- [ ] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by ayushi424**
<hr>

## Description 📜

I have added documentation for Regularization in Regression which includes code implementation with proper explanation for better understanding.
Fixes #4376  

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] Datascience with Python

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

### CODE REVIEW

**Comment by ayushi424** [comment]
Please add hacktoberfest accepted label so that this PR can be marked as a valid contribution for hacktoberfest 2021.

File: Datascience_With_Python/Machine Learning/Tutorials/Regularization in Regression/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# **Regularization in Regression**
+
+##  **Introduction**
+
+**What is Regularization ?**
+
+
+In mathematics, statistics, finance, computer science, particularly in machine learning and inverse problems, regularization is the process of adding information in order to solve an ill-posed problem or to prevent overfitting.
+
```
Comment (aishanii) [inline Datascience_With_Python/Machine Learning/Tutorials/Regularization in Regression/README.md:9]: This sentence is copied, please rephrase it.

File: Datascience_With_Python/Machine Learning/Tutorials/Regularization in Regression/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+## **Purpose**
+
+**Why do we need Regularization ?**
+
+Regularization is used in machine learning models to cope with the problem of overfitting i.e. when the difference between training error and the test error is too high.
+
```
Comment (aishanii) [inline Datascience_With_Python/Machine Learning/Tutorials/Regularization in Regression/README.md:15]: The first half of the sentence is copied please rephrase

File: Datascience_With_Python/Machine Learning/Tutorials/Regularization in Regression/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+- Lasso Regression
+- Ridge Regression
+
+**Lasso Regression**
+A regression model which uses L1 Regularization technique is called LASSO(Least Absolute Shrinkage and Selection Operator) regression.
+
+**Ridge Regression**
+A regression model that uses L2 regularization technique is called Ridge regression. 
+
```
Comment (aishanii) [inline Datascience_With_Python/Machine Learning/Tutorials/Regularization in Regression/README.md:31]: Rephrase this as well.

File: Datascience_With_Python/Machine Learning/Tutorials/Regularization in Regression/README.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+## **Explanation**
+
+
+**Regularization in Regression**
+There are mainly two types of regularization techniques:
+
+
+- Lasso Regression
+- Ridge Regression
+
```
Comment (aishanii) [inline Datascience_With_Python/Machine Learning/Tutorials/Regularization in Regression/README.md:25]: you can add a few lines about the differences between these two techniques

File: Datascience_With_Python/Machine Learning/Tutorials/Regularization in Regression/README.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+## **Usage**
+
+- Mainly used to deal with overfitting of data.
+- Improves score for training and testing data.
+
+## **Useful Resources to check out:**
+
+- https://towardsdatascience.com/regularization-in-machine-learning-76441ddcf99a
+- https://www.geeksforgeeks.org/regularization-in-machine-learning/
+- https://www.analyticsvidhya.com/blog/2021/05/complete-guide-to-regularization-techniques-in-machine-learning/
```
Comment (aishanii) [inline Datascience_With_Python/Machine Learning/Tutorials/Regularization in Regression/README.md:56]: add the roadmap of the implementation as well

File: Datascience_With_Python/Machine Learning/Tutorials/Regularization in Regression/regularization_in_regression.ipynb [hunk 6]
```diff
@@ -0,0 +1,706 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "R9hOOyJ4cVG0"
+      },
+      "source": [
+        "Here, we can see that after l2 regularization the scores for testing and training data have improved."
+      ]
```
Comment (aishanii) [inline Datascience_With_Python/Machine Learning/Tutorials/Regularization in Regression/regularization_in_regression.ipynb:703]: Add the important subheadings in bold for better visibility

**Review by aishanii — COMMENTED** [review]
please make the commented changes!

**Comment by ayushi424** [comment]
> please make the commented changes!

Done with changes. @aishanii @prathimacode-hub 
Since this PR was merged I have created an updated PR #4745  with commented changes.
